client/layout/masterbar/logged-out.jsx

```diff
@@ -30,6 +29,13 @@ function getLoginUrl( redirectUri ) {
 	return login( { ...params, isNative: config.isEnabled( 'login/native-login-links' ) } );
 }
 
+function getSignupUrl() {
+	const isJetpack =
+		typeof window !== 'undefined' &&
+		window.location.href.indexOf( 'jetpack%2Fconnect%2Fauthorize' ) >= 0;
```

Comment:
I suspect you could use the [`getInitialQueryArguments()`](https://github.com/Automattic/wp-calypso/blob/f2fda7c5eebe8e43f1c75a0529bcabde597cd628/client/state/selectors/get-initial-query-arguments.js#L18) or [`getCurrentQueryArguments()`](https://github.com/Automattic/wp-calypso/blob/f2fda7c5eebe8e43f1c75a0529bcabde597cd628/client/state/selectors/get-current-query-arguments.js#L18) selectors to retrieve the `redirect_to` url parameter - that should be working server-side as well, no?